lib/phoenix/channel.ex

```diff
@@ -137,14 +137,18 @@ defmodule Phoenix.Channel do
                                                                   :ignore |
                                                                   {:error, reason :: term, Socket.t}
 
-  defcallback leave(msg :: map, Socket.t) :: {:ok, Socket.t}
+  defcallback leave(msg :: map, Socket.t) :: :ok | {:error, reason :: term}
 
   defcallback handle_in(event :: String.t, msg :: map, Socket.t) :: {:ok, Socket.t} |
```

Comment:
Btw, we don't support `:ok`, so we should probably remove it. Also, I don't think we should support `:error`... but we can talk about it later. :)